url9d1719148e_mrqsdavi_pegmachine_tgz-pJ8-peg_RegexJ8/src/peg/Maquina.java

```diff
@@ -0,0 +1,234 @@
+package peg;
+import java.util.ArrayList;
+
+import Estruturas.Capture;
+import Instrucoes.Instrucao;
+import Instrucoes.TipoCapture;
+
+public class Maquina {
+
+	private char[] entrada;
+	private Instrucao instrucoes[];
+	EstadoMaquina estado;
+	boolean rodouEnd = false;
+	int tamanhoEntrada;
+	
+	public Maquina(String entrada, ArrayList<Instrucao> instrucoes){
+		Instrucao [] insArray = new Instrucao[instrucoes.size()];
+		insArray = instrucoes.toArray(insArray);
+		this.setEntrada(entrada.toCharArray());
+		this.setInstrucoes(insArray);
+		estado = new EstadoMaquina();
+		estado.inicializar();
+		estado.setI(0);
+		estado.setP(0);
+	}
+
+	public char[] getEntrada() {
+		return entrada;
+	}
+
+	public void setEntrada(char[] entrada) {
+		this.entrada = entrada;
+		tamanhoEntrada = entrada.length;
+	}
+
+	public Instrucao[] getInstrucoes() {
+		return instrucoes;
+	}
+
+	public void setInstrucoes(Instrucao[] instrucoes) {
+		this.instrucoes = instrucoes;
+	}
+	
+	public int getPosicaoEntrada(){
+		return estado.getI();
+	}
+	
+	public ArrayList<Capture> getCapturas(){
+		return estado.getCapturas();
+	}
+	
+	public void run(){
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 115 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 99% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test. We recommend that you simplify this method or break it into multiple methods.